ui/src/components/core/input/input.tsx

```diff
@@ -0,0 +1,43 @@
+import { Flex } from '../../layout';
+import { forwardRef } from 'react';
+import { StyledErrorMessage } from './input-error-message';
+import { StyledInputLabel } from './input-label';
+import { InputProps, StyledInput } from './input.styled';
+import { InputFileStyled } from './input-file';
+
+export const Input = forwardRef<InputProps, 'input'>((props, ref) => {
+  const {
+    label,
+    errorMessage,
+    rightIcon,
+    leftIcon,
+    type,
+    required,
+    ...ownProps
+  } = props;
+
+  return (
+    <Flex
+      css={{
+        flexDirection: 'column',
+        width: type === 'file' ? '$22' : ownProps.width,
+      }}
+    >
+      {label && (
+        <StyledInputLabel {...ownProps}>
+          {label}
+          {required && ' *'}
+        </StyledInputLabel>
+      )}
+      {type === 'file' ? (
+        <InputFileStyled {...ownProps} />
+      ) : (
+        <StyledInput ref={ref} {...ownProps} type={type} />

```

Comment:
Looking through the components usage, it seems like we need to add a pattern to component naming with prefix or suffix: `StyledInputLabel` `InputFileStyled`. One idea to keep this more organized is to wrap the components in a object or abstract class. For example, if you wrap all the styled components in an object inside `input.styled.ts` you would be able to locate all of them over an usage like: `StyledInput.Label` and `StyledInput.ErrorMessage`. Also we could use `Styled` only when we are declaring styling elements with stitches, `InputFileStyled` is a component that wraps logic, so it could be called just `InputFile`.

Reply:
Following up in the comment above, when wrapping the components in an object or abstract class:
- Objects are smaller structures in the memory than classes because of the prototype that is created;
- Using classes you are able to use namespaces to declare types, so you are able to define a class and a namespace both called with the same name, which improves the dev experience. Namespaces that have only type/interface declarations aren't included in the bundle, which is good to use only in this scenarios.